src/main/java/com/schibsted/spt/data/jslt/impl/BuiltinFunctions.java

```diff
@@ -377,21 +379,31 @@ public JsonNode call(JsonNode input, JsonNode[] arguments) {
     }
   }
 
-  // ===== HASH-CODE
 
-  public static class HashCode extends AbstractFunction {
+  // ===== SHA256
+
+  public static class Sha256 extends AbstractFunction {
 
-    public HashCode() {
-      super("hash-code", 1, 1);
+    public Sha256() {
+      super("sha256", 1, 1);
     }
 
     public JsonNode call(JsonNode input, JsonNode[] arguments) {
       // if input string is missing then we're doing nothing
       if (arguments[0].isNull())
         return arguments[0]; // null
 
-      String string = NodeUtils.toString(arguments[0], false);
-      return new IntNode(string.hashCode());
+      String message = NodeUtils.toString(arguments[0], false);
+      MessageDigest messageDigest;
+      try {
+        messageDigest = MessageDigest.getInstance("SHA-256");
+      } catch (NoSuchAlgorithmException e) {
+        throw new InternalError(e);
```

Comment:
Please throw `JsltException`. We use that for all exceptions thrown by JSLT, to signal that it was JSLT code that failed.

Reply:
Done